tests/test_webhooks.py

```diff
@@ -588,3 +590,125 @@ def test_get_remote_ip_remote_addr_is_none(self, data):
+
+        # add a secret to the webhook_endpoint
+        webhook_endpoint.secret = "whsec_rguCE5LMINfRKjmIkxDJM1lOPXkAOQp3"
+        webhook_endpoint.save()
+
+        # re-sync the WebhookEndpoint instance
+        WebhookEndpoint.sync_from_stripe_data(fake_webhook)
+
+        webhook_endpoint.refresh_from_db()
+        assert webhook_endpoint.id == fake_webhook["id"]
+        # assert secret persisted
+        assert webhook_endpoint.secret == "whsec_rguCE5LMINfRKjmIkxDJM1lOPXkAOQp3"
+
+        # assert UUID didn't get regenerated
+        assert webhook_endpoint.djstripe_uuid == djstripe_uuid
+
+    @pytest.mark.parametrize(
+        "data_secret_key",
+        [
+            None,
+            "",
+            "whsec_rguCE5LMINfRKjmIkxDJM1lOPXkAOQp3",
+        ],
+    )
+    @pytest.mark.parametrize(
+        "object_secret_key",
+        [
+            "",
+            "whsec_rguCE5LMINfRKjmIkxDJM1lOPXkAOQp3",
+        ],
+    )
+    def test__attach_objects_hook_object_exists_in_db(
+        self, data_secret_key, object_secret_key
+    ):
+        fake_obj_webhook = deepcopy(FAKE_WEBHOOK_ENDPOINT_1)
+        fake_webhook = deepcopy(FAKE_WEBHOOK_ENDPOINT_1)
+        # override the secret key
+        fake_obj_webhook["secret"] = object_secret_key
+
+        # create the WebHookEndpoint instance
+        webhook_endpoint = WebhookEndpoint.sync_from_stripe_data(fake_obj_webhook)
+
+        # override the secret key
+        fake_webhook["secret"] = data_secret_key
+        webhook_endpoint._attach_objects_hook(WebhookEndpoint, fake_webhook)
+
+        webhook_endpoint.save()
+
+        if object_secret_key:
+            assert webhook_endpoint.secret == object_secret_key
+        else:
+            assert webhook_endpoint.secret == data_secret_key
+
+    @pytest.mark.parametrize(
+        "data_secret_key",
+        [
+            None,
+            "",
+            "whsec_rguCE5LMINfRKjmIkxDJM1lOPXkAOQp3",
+        ],
+    )
+    def test__attach_objects_hook_object_does_not_exist_in_db(self, data_secret_key):
+        fake_webhook = deepcopy(FAKE_WEBHOOK_ENDPOINT_1)
+        # override the secret key
+        fake_webhook["secret"] = data_secret_key
+
+        # create the WebHookEndpoint instance
+        webhook_endpoint = WebhookEndpoint.sync_from_stripe_data(fake_webhook)
+
+        webhook_endpoint._attach_objects_hook(WebhookEndpoint, fake_webhook)
+
+        webhook_endpoint.save()
+        assert webhook_endpoint.secret == data_secret_key
+
+    def test___str__(self):
+        fake_webhook = deepcopy(FAKE_WEBHOOK_ENDPOINT_1)
+        webhook_endpoint = WebhookEndpoint.sync_from_stripe_data(fake_webhook)
+        assert (
+            str(webhook_endpoint)
+            == f"{webhook_endpoint.url}\n{webhook_endpoint.description}"
+        )
+        assert (
+            str(webhook_endpoint)
+            == "https://dev-example.com/stripe/webhook/1\nThis is the first webhook of a different website example.com"
```

Comment:
Let's not use anything other than `example.com`, `example.net` and `example.org` as domain names in tests, ever. If you want, you can use a subdomain eg `dev.example.com`.